includes/api/class-wc-admin-rest-reports-variations-controller.php

```diff
@@ -0,0 +1,264 @@
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $report, $request ) {
+		$data = $report;
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+		$response->add_links( $this->prepare_links( $report ) );
+
+		/**
+		 * Filter a report returned from the API.
+		 *
+		 * Allows modification of the report data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $report   The original report object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_report_variations', $response, $report, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param Array $object Object data.
+	 * @return array        Links for the given post.
+	 */
+	protected function prepare_links( $object ) {
+		$links = array(
+			'product'   => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d', $this->namespace, 'products', $object['product_id'] ) ),
+			),
+			'variation' => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d/%s/%d', $this->namespace, 'products', $object['product_id'], 'variation', $object['variation_id'] ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the Report's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'report_varitations',
+			'type'       => 'object',
+			'properties' => array(
+				'product_id'    => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Product ID.', 'wc-admin' ),
+				),
+				'variation_id'  => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Product ID.', 'wc-admin' ),
+				),
+				'items_sold'    => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Number of items sold.', 'wc-admin' ),
+				),
+				'gross_revenue' => array(
+					'type'        => 'number',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Total gross revenue of all items sold.', 'wc-admin' ),
+				),
+				'orders_count'  => array(
+					'type'        => 'integer',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'description' => __( 'Number of orders product appeared in.', 'wc-admin' ),
+				),
```

Comment:
Looks like that the schema doesn't include all data, even expanding, we should include some into the schema.

We probably could include an object here, like:

```json
{
    "product_id": 47,
    "variation_id": 357,
    "items_sold": 4,
    "gross_revenue": 120,
    "orders_count": 2,
    "extended_info": {
        "name": "T-Shirt",
        "price": "30",
        "image": "<img width=\"324\" height=\"324\" src=\"http://localhost/wp-content/uploads/2018/09/tshirt-2-324x324.jpg\" class=\"attachment-woocommerce_thumbnail size-woocommerce_thumbnail\" alt=\"\" srcset=\"http://localhost/wp-content/uploads/2018/09/tshirt-2-324x324.jpg 324w, http://localhost/wp-content/uploads/2018/09/tshirt-2-150x150.jpg 150w, http://localhost/wp-content/uploads/2018/09/tshirt-2-300x300.jpg 300w, http://localhost/wp-content/uploads/2018/09/tshirt-2-768x768.jpg 768w, http://localhost/wp-content/uploads/2018/09/tshirt-2-416x416.jpg 416w, http://localhost/wp-content/uploads/2018/09/tshirt-2-100x100.jpg 100w, http://localhost/wp-content/uploads/2018/09/tshirt-2.jpg 801w\" sizes=\"(max-width: 324px) 100vw, 324px\" />",
        "permalink": "http://localhost/product/t-shirt/",
        "attributes": {
            "color": "gray",
            "cut": "hipster cut"
        }
    }
}
```

So we can include all those items into the schema, and also return `extended_info` as an empty object when not requested by the filter.